### DISCUSSION TITLE
Why does an Enum with a single value get converted to a JSON schema `const`?

### DISCUSSION THREAD

**Discussion by jashandeep-sohi** [pydantic/pydantic#7441] [Question] [answered]
I'm curious why this is done? 
https://github.com/pydantic/pydantic/blob/c6e2f89f36805aef03194c366afb30faa798d188/pydantic/json_schema.py#L730

For context, I'm working on a FastAPI project. Converting it to a const (instead of enum) leads to confusion in the Swagger UI.

**Comment by jashandeep-sohi** [comment]
For example, the OpenAPI Guide uses an enum: https://swagger.io/docs/specification/describing-parameters/

**Comment by adriangb** [comment]
It can only ever have that one value, I think both options are correct. How does it cause confusion?

**Reply by jashandeep-sohi** [reply]
Agreed, logically it makes sense, but the Swagger UI won't render the field as a drop-down menu (as it does for Enums with multiple values).

I'm using the Enum field for a path parameter. Although it can only take on one value at the moment, I'd like to signal (to the API reader) that this parameter is semantically an enumeration of values and will be expanded eventually.

**Comment by jashandeep-sohi** [comment, marked as answer]
For anyone else looking for a workaround, I fixed my issue by mutating the JSON schema generation process:

```
from enum import StrEnum

from pydantic_core import CoreSchema

from pydantic import BaseModel, GetJsonSchemaHandler
from pydantic.json_schema import JsonSchemaValue

class ImageQuality(StrEnum):
    default = "default"

    @classmethod
    def __get_pydantic_json_schema__(
        cls,
        core_schema: CoreSchema,
        handler: GetJsonSchemaHandler
    ) -> JsonSchemaValue:
        json_schema = handler(core_schema)
        json_schema = handler.resolve_ref_schema(json_schema)
        json_schema.pop("const")

        json_schema["enum"] = [x.value for x in cls]

        return json_schema

```

You might need to play around with `x.name/x.value` depending on your configdict.

**Reply by MamfTheKramf** [reply]
Note that this can also be done for functionally created enums:

Edit: For dynamic Enums, however, you might want to check, that there is in fact only one variant. Otherwise you will get an exception at `json_schema.pop("const")`

```python
from enum import Enum
from pydantic_core import CoreSchema

from pydantic import BaseModel, GetJsonSchemaHandler
from pydantic.json_schema import JsonSchemaValue

variants = [("A", 0)]

MyEnum = Enum("MyEnum", variants)


@classmethod
def __get_pydantic_json_schema__(
    cls, core_schema: CoreSchema, handler: GetJsonSchemaHandler
) -> JsonSchemaValue:
    json_schema = handler(core_schema)
    json_schema = handler.resolve_ref_schema(json_schema)
    json_schema.pop("const")
    json_schema["enum"] = [x.value for x in cls]

    return json_schema

# equality is ok since enums with 0 variants will get an enum list as well
if len(MyEnum) == 1:
    setattr(MyEnum, "__get_pydantic_json_schema__", __get_pydantic_json_schema__)
```

**Comment by slafs** [comment]
We've just stumbled across this quirk too. And our reasoning/motivation is similar to the one @jashandeep-sohi mentioned:
> Although it can only take on one value at the moment, I'd like to signal (to the API reader) that this parameter is semantically an enumeration of values and will be expanded eventually.

Not a big deal, but surprising for sure 🤷‍♂️.
